src/NuGetForUnity/Editor/PackageContentManager.cs

```diff
@@ -194,12 +203,47 @@ internal static bool ShouldSkipUnpackingOnPath([NotNull] string path)
             return false;
         }
 
+        /// <summary>
+        ///     Extracts source files from a source code .nupkg <see cref="ZipArchive" /> into the <paramref name="baseDir" />/Sources.
+        /// </summary>
+        /// <param name="entries">The source file entries from the .nupkg zip file.</param>
+        /// <param name="baseDir">The path of the directory under which the 'Sources' subdirectory should be placed.</param>
+        internal static void ExtractPackageSources([NotNull] List<ZipArchiveEntry> entries, [NotNull] string baseDir)
+        {
+            var lastCommonDir = entries[0].FullName;
+            lastCommonDir = lastCommonDir.Substring(0, lastCommonDir.LastIndexOf('/') + 1);
+            for (var i = 1; i < entries.Count; ++i)
+            {
+                var entryFullName = entries[i].FullName;
+                for (var j = 0; j < entryFullName.Length && j < lastCommonDir.Length; j++)
+                {
+                    if (entryFullName[j] != lastCommonDir[j])
+                    {
+                        lastCommonDir = entryFullName.Substring(0, j);
+                        break;
+                    }
+                }
+            }
+
+            var sourcesDirectory = Path.Combine(baseDir, "Sources").Replace('\\', '/');
```

Comment:
Why do you need this Replace here? I don't see you use this variable later in a way that requiers it.

Reply:
It was useful when I was counting `/` occurrences, I will remove it now